py-server/main.py

```diff
@@ -84,6 +79,26 @@ def chat():
     return text, 200, {"Access-Control-Allow-Origin": "*"}
 
 
+def build_message_history(message_history_json):
+    """Parses client provided message history into ChatMessage objects
+
+    Args:
+        message_history_json (_type_): [ {content="xxx", author="AI|USER"} ]
+
+    Returns:
```

Comment:
nit: we should document under what conditions this function throws an exception

Reply:
done